spark/src/main/scala/org/apache/spark/sql/delta/PreprocessTableMerge.scala

```diff
@@ -119,14 +117,20 @@ case class PreprocessTableMerge(override val conf: SQLConf)
           target,
           m.resolvedActions,
           whenClauses = matched ++ notMatched ++ notMatchedBySource,
-          identityColumns = additionalColumns,
+          identityColumns = identityColumns,
           generatedColumns = generatedColumns,
           allowSchemaEvolution = withSchemaEvolution,
           postEvolutionTargetSchema = postEvolutionTargetSchema)
         m.copy(m.condition, alignedActions)
       case m: DeltaMergeIntoNotMatchedBySourceDeleteClause => m // Delete does not need reordering
     }
 
+    // A mapping from the identity column struct field to the GenerateIdentityColumnValues
+    // expression for the target table in the MERGE clause.
+    val identityColumnExpressionMap = mutable.Map[StructField, Expression]()
+    // Column names for which we need to track IDENTITY high water marks.
+    var trackHighWaterMarks = Set[String]()
+
```

Comment:
I'd either move these up to where `identityColumns` is defined to group them together OR I'd move these down to where they are first used. This is defined in an awkward place right now.

Reply:
Hmm they were defined here because they need to be outside the `notMatched.map {...}` block. Have moved them up.